clients/benchmarks/client.cpp

```diff
@@ -28,19 +29,58 @@
 #include "testing_geam.hpp"
 #include "testing_set_get_vector.hpp"
 #include "testing_set_get_matrix.hpp"
+#include "type_dispatch.hpp"
+
 #if BUILD_WITH_TENSILE
 #include "testing_gemm.hpp"
 #include "testing_gemm_strided_batched.hpp"
-#include "testing_gemm_kernel_name.hpp"
-#include "testing_gemm_strided_batched_kernel_name.hpp"
 #include "testing_trsm.hpp"
 #include "testing_gemm_ex.hpp"
 #include "testing_gemm_strided_batched_ex.hpp"
+
+// Template to dispatch testing_gemm_ex for performance tests
+// When Ti == void or complex, the test is marked invalid
+template <typename Ti,
+          typename To = Ti,
+          typename Tc = To,
+          typename    = typename std::conditional<!std::is_same<Ti, void>::value && !is_complex<Ti>,
+                                               std::true_type,
+                                               std::false_type>::type>
+struct perf_gemm_ex
+{
+    explicit operator bool() const { return true; }
+    void operator()(const Arguments& arg) { testing_gemm_ex<Ti, To, Tc>(arg); }
+};
+
+template <typename Ti, typename To, typename Tc>
+struct perf_gemm_ex<Ti, To, Tc, std::false_type> : rocblas_test_invalid
+{
+};
+
+// Template to dispatch testing_gemm_strided_batched_ex for performance tests
+// When Ti == void or complex, the test is marked invalid
+template <typename Ti,
+          typename To = Ti,
+          typename Tc = To,
+          typename    = typename std::conditional<!std::is_same<Ti, void>::value && !is_complex<Ti>,
```

Comment:
If this is C++14, you can write this as `std::is_same<Ti, void>{}` instead of `std::is_same<Ti, void>::value`.

Reply:
**Actually, that's allowed in C++11 as well.**

The syntactic sugar of `std::is_same<Ti, void>{}` or `std::is_same<Ti, void>()` (C++11) or `std::is_same_v<Ti, void>` (C++17) versus `std::is_same<Ti, void>::value` is not overly compelling.

The use of `<type_traits>` (`std::integral_constant`) classes were not introduced by this PR, and they were used with `::value` in the old C++11 code.